scripts/k8s_unit_windows.ps1

```diff
@@ -67,6 +67,23 @@ function Prepare-Vendor {
 
 }
 
+function Build-Kubeadm {
+    # For the cmd/kubeadm tests, we need to build the kubeadm binary and set the KUBEADM_PATH path.
+    # Before building the binary, we need to inject a few fields into k8s.io/component-base/version/base.go,
+    # otherwise version-related unit tests for kubeadm will fail. 
+    $buildFlags = @(
+        "-X 'k8s.io/component-base/version.gitTreeState=clean'",
+        "-X 'k8s.io/component-base/version.gitMajor=1'",
+        "-X 'k8s.io/component-base/version.gitMinor=27'",
+        "-X 'k8s.io/component-base/version.gitVersion=v1.27.0'",
+        "-X 'k8s.io/component-base/version.gitCommit=6a61da26b6761d1b86844cdec194ccaa02b41f3'"
```

Comment:
/lgtm
/hold

please remove hold when see fit.

these pinned versions should not cause problems, as long as kubedam unit tests don't expect a certain newer version to be present for example. but ideally the versions should come from git.

Reply:
also added one comment here about the revert PR:
https://github.com/kubernetes-sigs/windows-testing/pull/350#issuecomment-1344538942

Reply:
Right, there's currently no unit tests that expect a certain version to be set. And yes, ideally these versions should come from git.. but based on how https://github.com/kubernetes/kubernetes/blob/b30a6a3fc51715c8368281b09ecc6d29b1566069/staging/src/k8s.io/component-base/version/base.go#L42-L62 looks, that won't work 100%. We have ``gitVersion`` and ``gitCommit`` automatically set by git by running ``git archive`` as suggested in the comments there, but ``gitMajor``, ``gitMinor``, and ``gitTreeState`` will not be set. We have to set them ourselves. Of course, ``gitMajor`` and ``gitMinor`` are deprecated, and they'll be removed sometime in the future. We can remove them when that happens, and use ``git archive`` instead. We'd still have to set ``gitTreeState`` though.